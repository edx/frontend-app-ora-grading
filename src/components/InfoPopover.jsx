import React from 'react';
import PropTypes from 'prop-types';

import {
  OverlayTrigger,
  Popover,
  Icon,
  IconButton,
  PopoverContent,
} from '@edx/paragon';
import { InfoOutline } from '@edx/paragon/icons';

/**
 * <InfoPopover />
 */
export const InfoPopover = ({ children }) => (
  <OverlayTrigger
    trigger="focus"
    placement="auto"
    flip
    overlay={
      <Popover className="overlay-help-popover">
        <PopoverContent>{children}</PopoverContent>
      </Popover>
    }
  >
    <IconButton
      className="criteria-help-icon"
      src={InfoOutline}
      alt="criterion info"
      iconAs={Icon}
    />
  </OverlayTrigger>
);

InfoPopover.defaultProps = {};

InfoPopover.propTypes = {
  children: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.node),
    PropTypes.node,
  ]).isRequired,
};

export default InfoPopover;
