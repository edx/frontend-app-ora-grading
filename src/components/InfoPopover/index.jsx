import React from 'react';
import PropTypes from 'prop-types';

import {
  OverlayTrigger,
  Popover,
  Icon,
  IconButton,
} from '@edx/paragon';
import { InfoOutline } from '@edx/paragon/icons';
import { injectIntl, intlShape } from '@edx/frontend-platform/i18n';

import messages from './messages';

/**
 * <InfoPopover />
 */
export const InfoPopover = ({ children, intl }) => (
  <OverlayTrigger
    trigger="focus"
    placement="right-end"
    flip
    overlay={(
      <Popover className="overlay-help-popover">
        <Popover.Content>{children}</Popover.Content>
      </Popover>
    )}
  >
    <IconButton
      className="esg-help-icon"
      src={InfoOutline}
      alt={intl.formatMessage(messages.altText)}
      iconAs={Icon}
      onClick={() => {}}
    />
  </OverlayTrigger>
);

InfoPopover.defaultProps = {};
InfoPopover.propTypes = {
  children: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.node),
    PropTypes.node,
  ]).isRequired,
  intl: intlShape.isRequired,
};

export default injectIntl(InfoPopover);
