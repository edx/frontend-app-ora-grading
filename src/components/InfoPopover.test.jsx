import React from 'react';
import { shallow } from 'enzyme';

import InfoPopover from './InfoPopover';

jest.mock('@edx/paragon', () => ({
  OverlayTrigger: () => 'OverlayTrigger',
  Icon: () => 'Icon',
  IconButton: () => 'IconButton',
  Popover: () => 'Popover',
  PopoverContent: () => 'PopoverContent',
}));

jest.mock('@edx/paragon/icons', () => ({
  InfoOutline: jest.fn().mockName('icons.InfoOutline'),
}));

describe('Info Popover Component', () => {
  const child = <div>Children component</div>;
  test('snapshot', () => {
    expect(
      shallow(<InfoPopover>{child}</InfoPopover>),
    ).toMatchSnapshot();
  });

  describe('Component', () => {
    let el;
    beforeEach(() => {
      el = shallow(<InfoPopover>{child}</InfoPopover>);
    });
    test('Test component render', () => {
      expect(el.length).toEqual(1);
      expect(el.find('.criteria-help-icon').length).toEqual(1);
    });
  });
});
