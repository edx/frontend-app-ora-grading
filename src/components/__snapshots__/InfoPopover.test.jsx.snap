// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Info Popover Component snapshot 1`] = `
<OverlayTrigger
  flip={true}
  overlay={
    <Popover
      className="overlay-help-popover"
    >
      <PopoverContent>
        <div>
          Children component
        </div>
      </PopoverContent>
    </Popover>
  }
  placement="auto"
  trigger="focus"
>
  <IconButton
    alt="criterion info"
    className="criteria-help-icon"
    iconAs={[Function]}
    src={[Function]}
  />
</OverlayTrigger>
`;
